lib/web/device_trust.go

```diff
@@ -76,7 +81,44 @@ func (h *Handler) deviceWebConfirm(w http.ResponseWriter, r *http.Request, _ htt
 
 	// Always redirect back to the dashboard, regardless of outcome.
 	app.SetRedirectPageHeaders(w.Header(), "" /* nonce */)
-	http.Redirect(w, r, "/web", http.StatusSeeOther)
+
+	redirectTo, err := h.getRedirectURL(redirectURI)
+	if err != nil {
+		h.log.
+			WithError(err).
+			WithField("redirect_uri", redirectURI).
+			Debug("Unable to parse redirectURI")
+	}
+	http.Redirect(w, r, redirectTo, http.StatusSeeOther)
 
 	return nil, nil
 }
+
+// getRedirectURL tries to parse the given redirectURI. It will always return a redirect url
+// even if the parse fails (in case of failture, the returned string is "/web")
+func (h *Handler) getRedirectURL(redirectURI string) (string, error) {
```

Comment:
```suggestion
func (h *Handler) getRedirectPath(redirectURI string) (string, error) {
```

Since I guess technically this returns just the path?